openmdao/recorders/hdf5_recorder.py

```diff
@@ -63,6 +65,8 @@ def record_metadata(self, group):
 
         group = f.require_group('metadata')
 
+        group.create_dataset('om_case_version', data = om_case_version)
```

Comment:
Regarding (5), I agree that there will potentially be different version numbers for hdf5 and sqlite files, but it isn't necessary to put 'sqlite' or 'hdf5' in the name of the version variable because it will be obvious what you're talking about based on the context, i.e. you'll know that a file is a sqlite file or an hdf5 file, so putting that in the name is redundant. 

Putting 'om' in the name might give some people the impression that the version is the openmdao version, so I would recommend leaving that out.

My recommendation would be 'format_version'.

I'll shut up now, as this was a bit of a nit pick to start with.